### PR TITLE
Bug fix: Removing the sort function on destructive operations. 

### PR DESCRIPTION
**Pull request by Brendan-Reid1991**
[Issue link ](https://github.com/tqec/tqec/issues/406)

Small PR that ultimately removes a single line of code in `tqec/circuit/detectors/utils.py`. 

I modified the corresponding test in `utils_test.py`, by parameterizing the function and removing the calls to `set`, to ensure the ordering is maintained. 

All tests pass locally

The changes seem more significant than they are due to some black formatting that was done automatically when I saved the files. Also, I edited a function in `fragments.py` to use list comprehension rather than a loop. ~This is marked as a conflict, for some reason.~

### CODE REVIEW

**Comment by Brendan-Reid1991** [comment]
Converting to draft while I update my fork.

**Comment by nelimee** [comment]
Thanks for the PR! Note that this particular module (`tqec.circuit.detectors`) is currently being moved out of the `tqec` repository to https://github.com/tqec/tqecd. 

This fix will ultimately have to land in `tqecd`. Which option do you prefer:

1. You open a PR (nearly the same as this one) on the https://github.com/tqec/tqecd repository and I (or someone else) accept it.
2. I open a PR with your fix.

Option 1 requires a little bit more work from you, but you'll appear as the one that fixed that issue and as a contributor to `tqecd`. I am unsure how to guarantee you the same with Option 2 (i.e., that you will be seen as a contributor in the repository).

**Comment by nelimee** [comment]
Note 2: again, the fact that `CI/coverage-python` and `GitHub Pages/deploy` fail is not unexpected, and you should disregard these fails for the moment (it is a problem from the CI, not from your code). Issues with `CI/build-and-test-python` should still be fixed.

File: src/tqec/circuit/detectors/utils_test.py [hunk 1]
```diff
@@ -98,19 +105,20 @@ def test_iter_by_moment_repeat_block() -> None:
...
+    )
     # REPEAT block should NOT be with the last MR before it, even without TICK.
     assert len(list(iter_stim_circuit_by_moments(circuit))) == 5
 
 
-def test_collapse_pauli_strings_at_moment_all_basis() -> None:
-    qubits = list(range(5))
+@pytest.mark.parametrize("basis", ["", "X", "Y", "Z"])
+@pytest.mark.parametrize("qubits", [range(5), [1, 2, 0], [0, 1, 2], [2, 1, 0]])
+def test_collapse_pauli_strings_at_moment_all_basis(basis, qubits) -> None:
```
Comment (nelimee) [inline src/tqec/circuit/detectors/utils_test.py:116]:
```suggestion
def test_collapse_pauli_strings_at_moment_all_basis(basis: typing.Literal["", "X", "Y", "Z"], qubits: typing.Iterable[int]) -> None:
```
should fix the typing issue (not tested locally)

**Review by nelimee — COMMENTED** [review]
Started a review instead of just making a comment.

**Comment by Brendan-Reid1991** [comment]
went with option 1! Also included your comment on type declarations in the tests

[PR on TQECD ](https://github.com/tqec/tqecd/pull/6)

**Comment by nelimee** [comment]
Closing in favour of https://github.com/tqec/tqecd/pull/6

**Pull request closed without merging**